app/views.py

```diff
@@ -99,3 +104,61 @@ def send_email():
 @app.errorhandler(404)
 def page_not_found(error):
     return "oops.. page not found 😝"
+
+
+@app.route("/admin/services")
+def services_management():
+    services = services_model.all()
+    return render_template("admin/services/service_list.html", services=services)
+
+
+@app.route("/admin/services/adding", methods=["GET", "POST"])
+def services_adding():
+    if request.method == "POST":
+        images = None
+        if request.files.getlist("images"):
+            images = util.upload_files(
+                request.files.getlist("images"), app.config["IMAGES_DIR_PATH"]
+            )
+            images = [f"/static/images/{filename}" for filename in images]
+
+        price = [price.split(",") for price in request.form.getlist("price")]
+        services_model.insert(
+            name=request.form["name"],
+            price=price,
+            description=request.form["description"],
+            images=images,
+        )
+        return jsonify(data="OK")
+    return render_template("admin/services/service_upsert.html", mode="add")
+
+
+@app.route("/admin/services/<id>/editing", methods=["GET", "POST"])
```

Comment:
same note as above, the route names should not have `ing` also we should not handle both storing data and presenting it in one route